arbnode/transaction_streamer.go

```diff
@@ -1664,8 +1793,18 @@ func (s *TransactionStreamer) espressoSwitch(ctx context.Context, ignored struct
 		log.Error("ArbOS Config is nil")
 		return retryRate
 	}
-	if config.ArbitrumChainParams.EnableEspresso {
-		return s.submitEspressoTransactions(ctx, ignored)
+	if config.ArbitrumChainParams.EnableEspresso && s.config().SovereignSequencerEnabled {
```

Comment:
Should we delete `SovereignSequencerEnabled` here?

Reply:
For finality nodes, they shouldn't check the escape hatch, so I use this condition.

Reply:
Can you expand further on this? Our code will only run inside the batch poster, so we wanted to check the escape hatch condition. 

Which finality nodes are we referring to here?

Reply:
https://github.com/EspressoSystems/nitro-espresso-integration/blob/3248f9621ee418c0f3daf73a8706d66da5e3084c/system_tests/espresso_finality_node_test.go#L31

These code are in the tx streamer. So finality node in this test will also can run this function. I do think there should be a better way. I will try to refactor a bit

Reply:
Sorry that test is outdated, we will probably have to make changes to espresso finality node as well. So dont worry about the finality node right now. We will change it to not use `SovereignSequencerEnabled`

Reply:
Add a comment on it. We can do more clean up after most of the stuff having being done